### ISSUE TITLE
Ansible HVAC plugin fails to read if secrets is KV2

### ISSUE THREAD

**Issue by c0debreaker** [ansible/ansible#54462]
I'm using this code to pull secrets written in KV2.

```
prod_db: "{{ lookup('hashi_vault', 'secret=serverlab/production/db token=ce9c4734-4c71-971c-57d7-f5cd7240e860 url=https://127.0.0.1:8200 validate_certs=False')}}"
```

and I tried this too
```
prod_db: "{{ lookup('hashi_vault', 'secret=secret/serverlab/production/db token=ce9c4734-4c71-971c-57d7-f5cd7240e860 url=https://127.0.0.1:8200 validate_certs=False')}}"
```

However, I'm getting an error which is shown at the bottom of this ticket. It will be painful if we will have to duplicate all our secret keys. There are values that we don't know too. Is there something I am missing in Ansible that will tell it to use kv2?

I know hvac library can read kv2 since I wrote a python script that imports it. It was able to retrieve the secrets.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
- HVAC plugin

##### ANSIBLE VERSION

```
ansible 2.7.9
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.13 (default, Sep 26 2018, 18:42:22) [GCC 6.3.0 20170516]
```

##### CONFIGURATION

```
I ran "ansible-config dump --only-changed", but output was empty.
```

##### OS / ENVIRONMENT
- Linux debian 4.9.0-6-amd64 #1 SMP Debian 4.9.88-1+deb9u1 (2018-05-07) x86_64 GNU/Linux

##### STEPS TO REPRODUCE

```yaml
---

# Variables
#
# # Vault
debug:
prod_db: "{{ lookup('hashi_vault', 'secret=kv-v1/serverlab/production/db token=ce9c4734-4c71-971c-57d7-f5cd7240e860 url=https://127.0.0.1:8200 validate_certs=False')}}"
```

##### EXPECTED RESULTS
I was expecting the hvac plugin for Ansible will be able to retrieve the value of my secrets that were written in KV2. It's only working right now when I move my secrets to KV1.

##### ACTUAL RESULTS

```paste below
~/ansible ᐅ ansible-playbook --private-key /tmp/id_rsa_107 base.yml

PLAY [192.168.1.107] ********************************************************************************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************************************************************************
ok: [192.168.1.107]

TASK [base : Configure app] *************************************************************************************************************************************************************
/usr/local/lib/python2.7/dist-packages/urllib3/connectionpool.py:857: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
fatal: [192.168.1.107]: FAILED! => {"msg": "An unhandled exception occurred while templating '{{ lookup('hashi_vault', 'secret=serverlab/production/db token=ce9c4734-4c71-971c-57d7-f5cd7240e860 url=https://127.0.0.1:8200 validate_certs=False')}}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while running the lookup plugin 'hashi_vault'. Error was a <class 'ansible.errors.AnsibleError'>, original message: The secret serverlab/production/db doesn't seem to exist for hashi_vault lookup"}
	to retry, use: --limit @/root/ansible/base.retry

PLAY RECAP ******************************************************************************************************************************************************************************
192.168.1.107              : ok=1    changed=0    unreachable=0    failed=1   

```

**Comment by ansibot** [comment]
Files identified in the description:
None

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by c0debreaker** [comment]
!component

I don't understand the response. What is this?

**Comment by sivel** [comment]
Duplicate of https://github.com/ansible/ansible/pull/41132

**Comment by c0debreaker** [comment]
Just adding my solution here

I got hvac plugin working even without modification of the file `ansible/plugins/lookup/hashi_vault.py`.

I changed my template from
```
{
  "database": {
    "host": "{{ prod_db.host }}",
    "username": "{{ prod_db.username }}",
    "password": "{{ prod_db.password }}"
  }
}
```
to
```
{
  "database": {
    "host": "{{ prod_db.data.host }}",
    "username": "{{ prod_db.data.username }}",
    "password": "{{ prod_db.data.password }}"
  }
}
```

For kv1, the first json above will work and won't need to add `.data`.

My hashi_vault that worked for kv2 is `'secret=secret/data/serverlab/production/db`
For kv1, it should be `'secret=kv-v1/serverlab/production/db`

Of course, I created kv-v1 as kv1 engine in UI. I'm good! No need for updating `hashi_vault.py` file :D
